tests/ha/single_pod_restart/test_data_pod_restart.py

```diff
@@ -1372,195 +1372,230 @@ def test_pod_restart_node_nw_down(self):
-        pod_host = self.node_master_list[0].get_pod_hostname(pod_name=pod)
-
+        pod_name = random.sample(pod_list, 1)[0]
         loop_count = HA_CFG["common_params"]["loop_count"]
         for loop in range(1, loop_count):
             LOGGER.info("Running loop %s", loop)
-
-            LOGGER.info("Step 1: Shutdown the data pod by deleting deployment (unsafe)")
-            LOGGER.info("Deleting pod %s", pod)
-            resp = self.node_master_list[0].delete_deployment(pod_name=pod)
-            LOGGER.debug("Response: %s", resp)
-            assert_utils.assert_false(resp[0], f"Failed to delete pod {pod} by deleting deployment"
-                                               " (unsafe)")
-            LOGGER.info("Step 1: Successfully shutdown/deleted pod %s by deleting deployment "
-                        "(unsafe)", pod)
-            self.deployment_backup = resp[1]
-            self.deployment_name = resp[2]
-            self.restore_pod = True
-            self.restore_method = const.RESTORE_DEPLOYMENT_K8S
-
-            LOGGER.info("Step 2: Check cluster status")
-            resp = self.ha_obj.check_cluster_status(self.node_master_list[0])
-            LOGGER.debug("Response: %s", resp)
-            assert_utils.assert_false(resp[0], resp)
-            LOGGER.info("Step 2: Cluster is in degraded state")
-
-            LOGGER.info("Step 3: Check services status that were running on pod %s", pod)
-            resp = self.hlth_master_list[0].get_pod_svc_status(pod_list=[pod], fail=True,
-                                                               hostname=pod_host)
-            LOGGER.debug("Response: %s", resp)
-            assert_utils.assert_true(resp[0], resp)
-            LOGGER.info("Step 3: Services of pod are in offline state")
-
-            remain_pod_list = list(filter(lambda x: x != pod, pod_list))
-            LOGGER.info("Step 4: Check services status on remaining pods %s",
-                        remain_pod_list)
-            resp = self.hlth_master_list[0].get_pod_svc_status(pod_list=remain_pod_list,
-                                                               fail=False)
-            LOGGER.debug("Response: %s", resp)
-            assert_utils.assert_true(resp[0], resp)
-            LOGGER.info("Step 4: Services of pod are in online state")
-
-            LOGGER.info("Step 5: Create s3 account, create multiple buckets and run IOs")
-            users = self.mgnt_ops.create_account_users(nusers=1)
-            self.s3_clean.update(users)
-            self.test_prefix = 'test-34090'
+            LOGGER.info("Step 2: Shutdown %s data pod by deleting deployment and "
+                        "verify cluster & remaining pods status", pod_name)
+            resp = self.ha_obj.delete_kpod_with_shutdown_methods(
+                master_node_obj=self.node_master_list[0], health_obj=self.hlth_master_list[0],
+                down_method=const.RESTORE_DEPLOYMENT_K8S, delete_pod=pod_name)
```

Comment:
pod name will be different after every restart, deployment name will be same.
That needs to be taken care.

Reply:
ITs already taken care as last step in this loop:
  pod_name = self.node_master_list[0].get_recent_pod_name(
                deployment_name=self.deployment_name)